src/embedders/classification/contextual.py

```diff
@@ -155,39 +158,42 @@ def _encode(
                                     time.sleep(10.05)
                                 else:
                                     time.sleep(1)
-                        embeddings += [entry["embedding"] for entry in response["data"]]
+                        embeddings += [entry.embedding for entry in response.data]
                 else:
-                    response = openai.Embedding.create(
-                        input=documents_batch, engine=self.model_name
+                    response = self.openai_client.embeddings.create(
+                        input=documents_batch, model=self.model_name
                     )
-                    embeddings = [entry["embedding"] for entry in response["data"]]
+                    embeddings = [entry.embedding for entry in response.data]
                 yield embeddings
-            except openai_error.AuthenticationError:
+            except AuthenticationError:
                 raise Exception(
                     "OpenAI API key is invalid. Please provide a valid API key in the constructor of OpenAISentenceEmbedder."
                 )
 
+    @staticmethod
+    def load(embedder: dict) -> "OpenAISentenceEmbedder":
+        return OpenAISentenceEmbedder(
+            model_name=embedder["model_name"],
+            batch_size=embedder["batch_size"],
+            openai_api_key=embedder["openai_api_key"],
+            api_base=embedder["api_base"],
+            api_type=embedder["api_type"],
+            api_version=embedder["api_version"],
```

Comment:
might be mistaken but the flag for encoding (self.use_azure) isn't set, is this a problem?